### PR TITLE
LPD-21465 When filtering in management toolbar we should always set r…

### PR DESCRIPTION
**Pull request by antonio-ortega**
Hey,

This issue is quite well described [here](https://liferay.atlassian.net/browse/LPP-53343?focusedCommentId=2618997), specially in this sentence:

> What is happening is that when we filter, if the current page is higher than the results that we have after filtering, the results won’t appear, because in that page we don’t  have any results after filtering.

I think the only possibilities we have to fix this is either to set `resetCur` param to `true` or `cur` param to `1` so we always show the first page. Whichever param we update the result is the same.

However, I found this solution incredibly _hacky_ but I couldn't think about any other option, so you might have a different idea about how to approach this issue.

This could be very easily done by mgmt toolbar consumers, since they create those URLs and they'd just need to add such parameter, but I don't think this should be their responsability, and I think we should manage this internally inside ManagementToolbar component.

Please, let me know your thoughts.

Thanks!

### CODE REVIEW

**Comment by antonio-ortega** [comment]
ci:test:relevant

**Comment by antonio-ortega** [comment]
ci:test:relevant

**Comment by antonio-ortega** [comment]
ci:test:relevant

**Comment by bryceosterhaus** [comment]
> reset cur param to 1 so we always show the first page.

I think this is actually what I would expect any time we search/filter anything that could modify the total number of results.

I took a look at [MUI components](https://mui.com/x/) and they reset the page number when the "total" number could change. I believe github does this as well in issues and notifications.

https://github.com/liferay-frontend/liferay-portal/assets/6843530/96f6b76c-fe78-45bd-b356-d831716304fd

**Comment by antonio-ortega** [comment]
Thanks for your comment @bryceosterhaus ,

I guess you are agreeing with my solution since setting `resetCur` to `true` has same effect than setting `cur` to `1`, but let me know in case you are suggesting to change anything.

Unless there is any other comment, I'll forward this pull request.

Thanks.

Regards.

**Comment by antonio-ortega** [comment]
ci:test:relevant

**Comment by antonio-ortega** [comment]
ci:test:relevant

**Comment by antonio-ortega** [comment]
ci:test:sf

**Comment by antonio-ortega** [comment]
ci:test:sf

**Comment by antonio-ortega** [comment]
ci:test:sf

**Comment by antonio-ortega** [comment]
ci:test:relevant

**Comment by antonio-ortega** [comment]
ci:test:sf

**Comment by antonio-ortega** [comment]
ci:test:relevant

File: modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/management_toolbar/ManagementToolbar.js [hunk 1]
```diff
@@ -90,6 +90,38 @@ function ManagementToolbar({
...
+		filterDropdownItems.forEach((filterDropdownItem) => {
+			if (!filterDropdownItem.items) {
+				return null;
+			}
+
+			filterDropdownItem.items.forEach((item) => {
+				if (item.href) {
+					const urlParts = item.href.split('?');
+
+					const params = new URLSearchParams(urlParts[1]);
```
Comment (markocikos) [inline modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/management_toolbar/ManagementToolbar.js:107]: Using `URLSearchParams` works fine, but you might want to consider using `URL` API. It seems to me like a much more elegant way to manage parameters. [See here an example how we are using it](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/utils/index.js#L120-L135).

**Comment by antonio-ortega** [comment]
Thanks for your comments @markocikos ,

I'm closing this pull request since there are quite a few more cases we need to consider, so I'll resend a new pull request but I'll keep in mind your comments.

**Pull request closed without merging**